src/filesystem.ts

```diff
@@ -173,6 +169,13 @@ export class Filesystem {
     return link;
   }
 
+  private resolveLink(p: string) {
+    const symlink = fs.readlinkSync(p);
+    const parentPath = fs.realpathSync(path.dirname(p));
```

Comment:
Keep the comments to make it easier for other maintainers to understand, while also fixing issues with previously incorrect comments.

```suggestion
    // /var/tmp => /private/var
    const parentPath = fs.realpathSync(path.dirname(p));
```